fbw-a32nx/src/systems/instruments/src/MCDU/legacy_pages/A320_Neo_CDU_LateralRevisionPage.ts

```diff
@@ -40,11 +41,21 @@ export class CDULateralRevisionPage {
     }
     /** @type {BaseFlightPlan} */
     const targetPlan = inAlternate ? mcdu.getAlternateFlightPlan(forPlan) : mcdu.getFlightPlan(forPlan);
-
     const isPpos = leg === undefined || (legIndexFP === 0 && leg !== targetPlan.originLeg);
     const isFrom = legIndexFP === targetPlan.fromLegIndex && forPlan === FlightPlanIndex.Active && !inAlternate;
-    const isDeparture = legIndexFP === targetPlan.originLegIndex && !isPpos; // TODO this is bogus... compare icaos
-    const isDestination = legIndexFP === targetPlan.destinationLegIndex && !isPpos; // TODO this is bogus... compare icaos
+    const legWaypoint = !isPpos ? leg.definition.waypoint : null;
+    const departure = targetPlan.originAirport;
+    const destination = targetPlan.destinationAirport;
+    const isDeparture =
+      legIndexFP === targetPlan.originLegIndex &&
+      !isPpos &&
+      isAirport(legWaypoint) &&
```

Comment:
```suggestion
      (isAirport(legWaypoint) || isRunway(legWaypoint)) &&
```
would address the comment. You probably have to write isRunway. Type isAirport should probably go in Airport.ts and isRunway in Runway.ts.